gwnr/waveform/esigma_utils.py

```diff
@@ -579,132 +793,98 @@ def get_imr_esigma_modes(
-                in the orbital frequency array of length {len(orb_freq)}"""
-            )
-
-        if keep_f_mr_transition_at_center and not (
-            failsafe
-            and ((orb_phase[transition_idx] + num_hyb_orbits * np.pi) > orb_phase[-1])
-        ):
-            # Orbital cycle based hybridization that keeps f_mr_transition at
-            # the hybridization frequency window's midpoint
-            if (orb_phase[transition_idx] + num_hyb_orbits * np.pi) > orb_phase[-1]:
-                raise Exception(
-                    f"""Requested number of orbits to hybridize over not available in the waveform after the transition frequency.
-Either decrease the number of orbits to hybridize over (currently {num_hyb_orbits}) or decrease the inspiral-to-merger transition frequency."""
-                )
-
-            window_start_idx = (
-                np.searchsorted(
-                    orb_phase, orb_phase[transition_idx] - num_hyb_orbits * np.pi
-                )
-                - 1
-            )  # phase is always a monotonically increasing function, hence using the more efficient np.searchsorted method
-            window_end_idx = np.searchsorted(
-                orb_phase, orb_phase[transition_idx] + num_hyb_orbits * np.pi
-            )
-            f_window_mr_transition = (
-                2
-                * np.max(
-                    [
-                        orb_freq[window_end_idx] - f_mr_transition / 2.0,
-                        f_mr_transition / 2.0 - orb_freq[window_start_idx],
-                    ]
-                )
-                * 2
-            )  # Extra 2-factor for returning the (2,2)-mode frequency
-        else:
-            # Orbital cycle based hybridization that keeps f_mr_transition at the hybridization frequency window's end
-            window_start_idx = (
-                np.searchsorted(
-                    orb_phase, orb_phase[transition_idx] - 2 * np.pi * num_hyb_orbits
-                )
-                - 1
-            )
-            f_window_mr_transition = 2 * (
-                orb_freq[transition_idx] - orb_freq[window_start_idx]
-            )  # Extra 2-factor for returning the (2,2)-mode frequency
-
-            if verbose > 4:
-                print(
-                    f"""The transition is to happen in a window from
-                    frequencies: [{orb_freq[window_start_idx]}, {orb_freq[transition_idx]}]Hz,
-                    between indices: [{window_start_idx}, {transition_idx}]"""
-                )
-        if verbose > 4:
-            print(f"""f_window_mr_transition = {f_window_mr_transition}""")
+        f_window_mr_transition = _get_transition_frequency_window(
+            retval[-2]["phi"],
+            orbital_frequency,
+            delta_t,
+            f_mr_transition,
+            num_hyb_orbits,
+            keep_f_mr_transition_at_center,
+            hybridize_using_orbital_frequency,
+            failsafe=True,
```

Comment:
`failsafe` can be removed here I guess (see my comment on the code of `_get_transition_frequency_window`)

Reply:
fixed